File: backend/routes/generate.py
from fastapi import APIRouter
from pydantic import BaseModel
from services.ai_client import ask_ai
from services.prompts import PROMPTS

router = APIRouter()

class GenerateRequest(BaseModel):
    mode: str
    input: dict

@router.post("/api/generate")
def generate(req: GenerateRequest):
    system_prompt = PROMPTS.get(req.mode, "You are a helpful assistant.")
    user_prompt = str(req.input)
    result = ask_ai(system_prompt, user_prompt)
    return {"result": result}

File: frontend/src/api/generate.js
import axios from "axios";

const API_URL = import.meta.env.VITE_API_URL || "http://127.0.0.1:8000";

export async function generate(mode, input) {
  const res = await axios.post(`${API_URL}/api/generate`, { mode, input });
  return res.data.result;
}